libs/wingsdk/src/target-tf-aws/api.ongetrequest.inflight.ts

```diff
@@ -0,0 +1,25 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import type { IFunctionHandlerClient } from "../cloud";
+import type { IApiGetHandlerHandlerClient } from "../cloud/api";
+
+type IHandler = IFunctionHandlerClient<APIGatewayProxyEvent, string>;
+
+export class ApiOnGetRequestHandlerClient
+  implements
+    IApiGetHandlerHandlerClient<APIGatewayProxyEvent, APIGatewayProxyResult>
+{
+  private readonly handler: IHandler;
+  constructor({ handler }: { handler: IHandler }) {
+    this.handler = handler;
+  }
+  public async handle(
+    request: APIGatewayProxyEvent
+  ): Promise<APIGatewayProxyResult> {
+    const resp = await this.handler.handle(request);
+    // TODO: this is just a WIP to get the API working, we need to figure out what needs to be returned by the inflight function
```

Comment:
I believe the [spec] stipulates this should be an `ApiResponse` struct.

[spec]: https://docs.winglang.io/reference/wingsdk-spec#api

Reply:
yes you are right.
But I have problem here as well. 
We try to describe the api request on a cloud independent "language". That means we say for example that the response has a key `status` whereas when implemented the aws apigateway expect a key `statusCode`. 
I am not sure how to model that with types? 
If you look at this handle method. It should implement the specification from the cloud spec. But in the real world the request does look different. On which level do I map the `APIGatewayProxyEvent` to the `ApiRequest` from the cloud spec?

If I change the handle method to align with the `ApiRequest` interface this will be not work in the real world because the apigateway in aws creates a  `APIGatewayProxyEvent`.

Reply:
Have a look at this commit https://github.com/winglang/wing/pull/1285/commits/8a16798f032af234ee5f769c78a7f34dc8161de1 in file 
libs/wingsdk/src/target-tf-aws/api.onrequest.inflight.ts

What do you think?

Reply:
Looks great 👍